packages/react-components/react-combobox/src/components/Option/Option.types.ts

```diff
@@ -18,9 +18,17 @@ export type OptionProps = ComponentProps<Partial<OptionSlots>> & {
    */
   disabled?: boolean;
 
+  /**
+   * A string used as the option's display value if no children are present.
+   * This is used as the Dropdown button's or Combobox input's value when the option is selected,
+   * and as the comparison for type-to-find keyboard functionality.
+   */
+  label: string;
```

Comment:
What if we make `label` required _only if_ `children` is not a string? I'm not 100% sure I like this idea, but I wanted to throw it out there. (* And in this case, I think I prefer to call the prop `text` or `plainText`, but that's less important.)

As an experiment, I tried making the props type be:

```ts
export type OptionProps = ComponentProps<Partial<OptionSlots>> & {
  disabled?: boolean;
  value?: string;
} & (
    | {
        text?: string;
        children: string;
      }
    | {
        text: string;
        children?: React.ReactNode;
      }
  );
```

Some examples of how it works:

```jsx
<>
  {/* ✅ Ok: string children */}
  <Option>hello</Option>

  {/* ✅ Ok: using text prop */}
  <Option text="hello" />

  {/* ❌ Error: Property 'text' is missing in type '{ children: Element; }' but required in type '{ text: string; children?: ReactNode; }' */}
  <Option>
    <b>hello</b>
  </Option>

  {/* ✅ Ok: non-string children, but also providing text prop */}
  <Option text="hello">
    <b>hello</b>
  </Option>

  {/* ✅ Ok: It's fine to have a plain string for both */}
  <Option text="hello">hi</Option>

  {/* ❌ Error:  Type '{}' is not assignable to type 'IntrinsicAttributes & (OptionProps & RefAttributes<HTMLDivElement>)' */}
  {/* It's pretty cryptic 🫤, but happens because at least one of 'text' or 'children' is required. */}
  <Option />
</>
```

* **Pros**
  * More ergonomic in the simple case (string content can just be children: `<Option>cat</Option>`)
  * Requires fewer updates to places that were using the [unstable] `<Option>` component. Only places that use JSX as children would need to be updated.
  * Compile-time error if there's no plain text label provided.
* **Cons**
  * Advanced TypeScript tends to be confusing or hard to understand.
  * Surprising error if you change children from a plain string to JSX.
  * Long chain of defaults: `value` defaults to `text`, and `text` defaults to `children`.
  * More than one way to do the same thing (either use `text` prop, or provide children)

Reply:
I do like this approach :).

Can you think of any issues where it might make the props type too complex, or hard to extend? That would be my main worry, since we already have pretty convoluted prop types 🙈

Reply:
we can always loosen the type later and support this in future releases.